core/src/modules/ops/destination.ts

```diff
@@ -605,7 +605,7 @@ export namespace DestinationOps {
     });
 
     for (const _export of _exports) {
-      const record = records.find((r) => (r.id = _export.recordId));
+      const record = records.find((r) => r.id === _export.recordId);
```

Comment:
This was the problem

Reply:
Hmm- I wonder how this didn't fail any other tests. I believe this would have been finding the first record regardless of its ID

Reply:
We don't look at the record's ID... ever... in the destinations.  It's something I noticed specifically in the logger, that's why I added a test there for it